src/main/java/net/minestom/server/instance/InstanceContainer.java

```diff
@@ -488,7 +488,7 @@ private void executeNeighboursBlockPlacementRule(@NotNull Point blockPosition) {
         for (int offsetX = -1; offsetX < 2; offsetX++) {
             for (int offsetY = -1; offsetY < 2; offsetY++) {
                 for (int offsetZ = -1; offsetZ < 2; offsetZ++) {
-                    if (offsetX == 0 && offsetY == 0 && offsetZ == 0)
+                    if (offsetX == 0 && offsetY == 0 && offsetZ == 0 || offsetY < getDimensionType().getMinY() || offsetY > getDimensionType().getTotalHeight())
```

Comment:
I believe you need to compare against the neighborY number, you're just comparing against the offset, not the true position

Reply:
you're right, my bad